src/System.Private.CoreLib/shared/System/Buffers/ArrayPool.cs

```diff
@@ -33,7 +37,7 @@ public abstract class ArrayPool<T>
         /// optimized for very fast access speeds, at the expense of more memory consumption.
         /// The shared pool instance is created lazily on first access.
         /// </remarks>
-        public static ArrayPool<T> Shared { get; } = new TlsOverPerCoreLockedStacksArrayPool<T>();
+        public static ArrayPool<T> Shared { get; } = s_shared;
```

Comment:
Does this need to be just `public static ArrayPool<T> Shared => s_shared;`? 

I expect that `Shared { get; } =` will create a hidden backing field of the wrong type.

Reply:
Ha! Definately

Reply:
Yes...